### ISSUE TITLE
Clarify when to use NDARRAY as data type

### ISSUE THREAD

**Issue by philip-khor** [KonduitAI/konduit-serving-docs#1]
TL;DR It is not clear when to use NDARRAYs as data types, and why. 

In the documentation, we list a set of possible data types, of which NDARRAY is one. There are several problems with the current approach: 

1. NDARRAY is not always acceptable. For example, TensorFlow steps accept a limited set of values, and NDARRAY is not one of these (org.nd4j.tensorflow.conversion.TensorDataType). 
2. It's not clear what happens when in a Python step, what happens if a non-NDARRAY data type is used for a NUMPY-formatted input. Presumably the scalar type of the NumPy arrays are cast, but this should be explicit.  

I don't have a clear action plan in mind, but eventually it should be clear to the user what format-type combinations should be used for their use case.
